crates/accelerate/src/sabre_swap/mod.rs

```diff
@@ -252,7 +320,7 @@ fn swap_map_trial(
     heuristic: &Heuristic,
     seed: u64,
     mut layout: NLayout,
-) -> TrialResult {
+) -> (SabreResult, NLayout) {
```

Comment:
Minor: This function always returns the `Layout` that's mutated via the `mut layout` field, I think?  If so, it might be clearer to not return the object and force the caller to store the output if they need it later - I got tripped up reading code further down that passed in a `layout.copy()` and then accepted an `NLayout` from the output; it _looks_ like it allocates two `NLayout` objects rather than one.

Reply:
Yeah, I found that to be odd as well, but left it as is. I suppose what we want here is to accept `layout` as a mutable reference, rather than moving from the caller and then moving back to the caller.

Reply:
Ohhh, this is partially my mistake. I hadn't realised that the code took ownership of `layout` - I think it was an `&mut` parameter, but hadn't read it correctly.

In this case, it's probably not worth bothering changing the API right now, since it's effectively pre-existing, just hidden in the `TrialResult` struct before.